packages/myst-directives/src/table.ts

```diff
@@ -2,6 +2,54 @@ import type { DirectiveSpec, DirectiveData, GenericNode } from 'myst-common';
 import { fileError, normalizeLabel, RuleId } from 'myst-common';
 import type { VFile } from 'vfile';
 
+export const tableDirective: DirectiveSpec = {
+  name: 'table',
+  arg: {
+    type: 'myst',
+  },
+  options: {
+    label: {
+      type: String,
+      alias: ['name'],
+    },
+    class: {
+      type: String,
+      // class_option: list of strings?
+      doc: `CSS classes to add to your table. Special classes include:
+
+- \`full-width\`: changes the table environment to cover two columns in LaTeX`,
+    },
+    align: {
+      type: String,
+      // choice(['left', 'center', 'right'])
+    },
+  },
+  body: {
+    type: 'myst',
+    required: true,
+  },
+  run(data: DirectiveData): GenericNode[] {
+    const children = [];
+    if (data.arg) {
+      children.push({
+        type: 'caption',
+        children: [{ type: 'paragraph', children: data.arg as GenericNode[] }],
+      });
+    }
+    children.push(...(data.body as GenericNode[]));
```

Comment:
Could add validation here that children includes a `table` node? 🤷‍♀️ Trade-off between strictness now vs. unexpected behaviour later.

Looks like in other directives we err towards less strict, so maybe best to leave as-is.

Reply:
I did think about that as well. Probably a good idea to add it and at least warn.